modules/ting_search_carousel/ting_search_carousel.module

```diff
@@ -283,31 +283,30 @@ function _ting_search_carousel_find_entities_with_covers($query, $start, $size,
   $finished = FALSE;
   $entities = array();
 
-  $options = array(
-    'facets' => array(),
-    'enrich' => TRUE,
-    'allObjects' => FALSE,
-  );
+  $sal_query = ting_start_query()
+    ->setRawQuery($query)
+    ->setPage($start)
+    ->setCount($size)
+    ->setPopulateCollections(FALSE)
+    ->setMaterialFilter($ignore, FALSE);
 
-  // TODO: use the Search Abstraction Layer (ting_start_query()).
-  // Ting pages are 1, not 0, based.
-  $result = opensearch_do_search($query, ($start++) + 1, $size, $options);
+  $results = $sal_query->execute();
+  $collections = $results->getTingEntityCollections();
```

Comment:
I would expect `$results->getTingEntityCollections()` to be moved to `L305`.

The check on `L297` does not make any sense if we already assume on `L294` that the return value from `$sal_query->execute()` is a proper object.